src/scipp/core/concepts.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
The name `concepts` does not really tell me what is inside the file. These functions are pretty diverse. So I can't really think of a great name, but I would go with something like `util`.

Reply:
It is *not* meant to be a generic `util` file, rather a collection of functions that encapsulate key concepts (behaviors) of Scipp. Is `behaviors` a better name?

Reply:
I notice I *did* place a util function here by accident, I moved it to `util.py`.

Reply:
To me both `concepts` and `behaviors` sounds like it contains base classes or protocols or something similar.